### DISCUSSION TITLE
Gatsby Incremental builds

### DISCUSSION THREAD

**Discussion by mustofa-binary** [vercel/vercel#5001] [Help] [answered]
I can't find a way on how to do real gatsby incremental builds on vercel. i saw only netlify and gatsby cloud that is currently support it. can we have that feature as well in vercel. it improves the build time to 80-90% which helps development a lot!. i hope this feature is considered 🙏

**Comment by styfle** [comment]
I believe this already works on Vercel. Can you share an example that isn't working correctly?

**Reply by mustofa-binary** [reply]
may you share how to do it on vercel though, what we did was just changing gatsby build command to 
```
GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES=true gatsby build --log-pages
```

but it seems that there is no difference between the builds with normal gatsby build command. it tooks around 12-14 minutes for the build. when i try to deploy to gatsby cloud it only took me 1 or 2 minutes to deploy. are there more steps i need to do to enable incremental bulids on vercel?

**Comment by styfle** [comment]
We already persist the `.cache` directory and `public` directory between builds.

But you're right Gatsby Incremental Builds doesn't seem to work--modifying a single page rebuilds all the pages.

I couldn't even get it to work locally on my machine 🤔 

```
npx gatsby-cli new gatsby-example
cd gatsby-example
yarn install
yarn build
```

Then I updated `package.json` build script from `gatsby build` to `GATSBY_EXPERIMENTAL_PAGE_BUILD_ON_DATA_CHANGES=true gatsby build --log-pages`.

When I ran `yarn build` a couple times after changing a single file, I still see all pages were updated:

```
Updated page: /404/
Updated page: /
Updated page: /page-2/
Updated page: /using-typescript/
Updated page: /404.html
```

Seems like a Gatsby bug to me.

**Reply by mustofa-binary** [reply]
yup thats what happen to me too, i wonder how come it works on gatsby cloud and netlify though

**Reply by johnkors** [reply]
Did you find how to get this working on Vercel, @mustofa-binary ?

**Reply by teug91** [reply]
Same here. Does not work locally or on Vercel. Works on Netlify and Gatsby Cloud.

**Reply by styfle** [reply]
Let's see what the Gatsby team has to say about this: https://github.com/gatsbyjs/gatsby/issues/26741

**Reply by johnkors** [reply]
Thanks, @styfle 👍

**Comment by styfle** [comment, marked as answer]
Here's the official response from the Gatsby team:

> Conditional Page builds is an experimental feature that only works with data updates. Data updates in Gatsby means, data that comes from the GraphQL layer, in this case only page queries are supported. It can work with JSON updates when using `gatsby-transformer-json` and `gatsby-source-filesystem`.
> 
> Gatsby at this point in time is not going to invest in this feature and it will probably stay experimental without any development.

https://github.com/gatsbyjs/gatsby/issues/26741#issuecomment-685416917

**Reply by teug91** [reply]
We are experiencing this issue even though the source code remains unchanged from the previous deployment.
